### PR TITLE
Build index name with template.

### LINKED ISSUE #1: More flexible index pattern
These tests assume that indices are namespaced by org and space. This may not always be the case--actually, @cnelson has found that that index strategy results in unusably bad performance on our cluster. These tests should expose a more flexible method to configure search index. One option might be go templates:

```json
{
  "elasticsearch_app_index": "logs-app-{{org}}-{{date}}"
}
```

If that makes sense, I'd be happy to send a patch.

### PR DESCRIPTION
**Pull request by jmcarp**
Resolves #1.

### CODE REVIEW

File: smoke/smoke_test.go [hunk 1]
```diff
@@ -26,10 +28,26 @@ var _ = Describe("Logsearch", func() {
...
+	elasticIndex := func(raw string) (string, error) {
+		tpl, err := template.New("index").Parse(raw)
+		var buf bytes.Buffer
+		if err != nil {
+			return "", err
+		}
+		if err := tpl.Execute(&buf, map[string]interface{}{
+			"org":   ctx.RegularUserContext().Org,
+			"space": ctx.RegularUserContext().Space,
+			"time":  time.Now().Local(),
```
Comment (Infra-Red) [inline smoke/smoke_test.go]: Fields org,space,time for template should be capitalized.

File: smoke/smoke_test.go [hunk 2]
```diff
@@ -42,23 +60,17 @@ var _ = Describe("Logsearch", func() {
...
+		index, err := elasticIndex(config.ElasticsearchAppIndex)
+		Expect(err).NotTo(HaveOccurred())
 
-		testUri := fmt.Sprintf(
-			elaticUri(elasticEndpoint)+"/"+config.ElasticsearchAppIndex+
-				"-"+ctx.RegularUserContext().Org+
-				"-"+ctx.RegularUserContext().Space+
-				"-"+"%v"+"/_search?q=@cf.app:"+"%s",
-			current_time.Format("2006.01.02"), appName)
+		testURI := elasticURI(elasticEndpoint) + "/" + index + "_search?q=@cf.app:" + appName
```
Comment (Infra-Red) [inline smoke/smoke_test.go]: Slash is missing before search query.

File: README.md [hunk 3]
```diff
@@ -32,6 +32,7 @@ Below is an example `integration_config.json`:
   "system_domain":                    "example.com",
   "admin_user":                       "admin",
   "admin_password":                   "admin",
+  "elasticsearch_app_index":          "logs-app-{{.Org}}-{{.Space}}-{{.Time.Format \"2006-01-02\"}}",
```
Comment (Infra-Red) [inline README.md]: By default timestamp will be created with dots instead of dashes.

**Review by Infra-Red — CHANGES_REQUESTED** [review]
Hi @jmcarp, here the couple of changes required before merging this PR.

**Comment by jmcarp** [comment]
Thanks for catching those mistakes @Infra-Red. Typos should be fixed now.

**Review by Infra-Red — APPROVED** [review]

**Comment by Infra-Red** [comment]
Thanks @jmcarp

**Pull request merged**